packages/client-sdk-nodejs/src/internal/data-client.ts

```diff
@@ -2563,6 +2565,155 @@ export class DataClient implements IDataClient {
+      cacheName,
+      this.convert(sortedSetName)
+    );
+
+    this.logger.trace(
+      "'sortedSetLength' request result: %s",
+      truncateString(result.toString())
+    );
+    return result;
+  }
+
+  private async sendSortedSetLength(
+    cacheName: string,
+    sortedSetName: Uint8Array
+  ): Promise<CacheSortedSetLength.Response> {
+    const request = new grpcCache._SortedSetLengthRequest({
+      set_name: sortedSetName,
+    });
+
+    const metadata = this.createMetadata(cacheName);
+    return await new Promise(resolve => {
+      this.clientWrapper.getClient().SortedSetLength(
+        request,
+        metadata,
+        {
+          interceptors: this.interceptors,
+        },
+        (err, resp) => {
+          if (resp?.missing) {
+            resolve(new CacheSortedSetLength.Miss());
+          } else if (resp?.found) {
+            if (!resp.found.length) {
+              resolve(new CacheSortedSetLength.Miss());
+            } else {
+              resolve(new CacheSortedSetLength.Hit(resp.found.length));
+            }
+          } else {
+            resolve(
+              new CacheSortedSetLength.Error(cacheServiceErrorMapper(err))
+            );
+          }
+        }
+      );
+    });
+  }
+
+  public async sortedSetLengthByScore(
+    cacheName: string,
+    sortedSetName: string,
+    scoreRange?: {
+      minScore?: number;
+      maxScore?: number;
+    }
+  ): Promise<CacheSortedSetLengthByScore.Response> {
+    try {
+      validateCacheName(cacheName);
+      validateSortedSetName(sortedSetName);
+      if (scoreRange)
+        validateSortedSetScores(scoreRange.minScore, scoreRange.maxScore);
+    } catch (err) {
+      return new CacheSortedSetFetch.Error(normalizeSdkError(err as Error));
+    }
+
+    this.logger.trace(
+      "Issuing 'sortedSetLengthByScore' request; minScore: %s, maxScore: %s",
+      scoreRange?.minScore?.toString() ?? 'null',
+      scoreRange?.maxScore?.toString() ?? 'null'
+    );
+
+    const result = await this.sendSortedSetLengthByScore(
+      cacheName,
+      this.convert(sortedSetName),
+      scoreRange
+    );
+
+    this.logger.trace(
+      "'sortedSetLengthByScore' request result: %s",
+      truncateString(result.toString())
+    );
+    return result;
+  }
+
+  private async sendSortedSetLengthByScore(
+    cacheName: string,
+    sortedSetName: Uint8Array,
+    scoreRange?: {
+      minScore?: number;
+      maxScore?: number;
+    }
+  ): Promise<CacheSortedSetLengthByScore.Response> {
+    const request = new grpcCache._SortedSetLengthByScoreRequest({
+      set_name: sortedSetName,
+    });
+
+    if (scoreRange?.minScore) {
+      request.inclusive_min = scoreRange.minScore;
+    } else {
+      request.unbounded_min = new grpcCache._Unbounded();
+    }
+
+    if (scoreRange?.maxScore) {
+      request.inclusive_max = scoreRange.maxScore;
+    } else {
+      request.unbounded_max = new grpcCache._Unbounded();
+    }
```

Comment:
Subtle bug here: in JavaScript, `undefined` and `0` both evaluate to `false`. Thus with the code as-is, if a developer passes `minScore` (`maxScore`) as `0`, this block will then set the request `min` (`max`) to `Unbounded`, which we don't want.

Instead test explicitly for `undefined`:

```suggestion
    if (scoreRange?.minScore === undefined) {
      request.unbounded_min = new grpcCache._Unbounded();
    } else {
      request.inclusive_min = scoreRange.minScore;
    }

    if (scoreRange?.maxScore === undefined) {
      request.unbounded_max = new grpcCache._Unbounded();
    } else {
      request.inclusive_max = scoreRange.maxScore;
    }
```

You can see examples of this elsewhere in this file :)